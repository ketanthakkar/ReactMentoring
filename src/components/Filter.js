import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { sortMovies, fetchMovies } from '../actions/index'
import PropTypes from 'prop-types';

class Filter extends Component {

    handleSortClick = (event) => {
        const sortBy = event.target.id === "release" ? "release_date" : "vote_average";
        this.props.sortMovies(sortBy);
        this.props.fetchMovies();

        document.getElementById("release").classList.toggle("highlight-color");
        document.getElementById("rating").classList.toggle("highlight-color");
      };

    render() {
        return (
            <section className="filter-container">
                <span className="results">{this.props.movieCount} movies found</span>
                <div className="sort-detail">
                    <span>Sort by</span>
                    <span id="release" className="highlight-color" onClick={this.handleSortClick}>release date</span>
                    <span id="rating" className="" onClick={this.handleSortClick}>rating</span>
                </div>
            </section>
        )
    }
}

Filter.propTypes = {
    movieCount: PropTypes.number
};

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
      sortMovies,
      fetchMovies
    }, dispatch);
  }

export default connect(null, mapDispatchToProps)(Filter);