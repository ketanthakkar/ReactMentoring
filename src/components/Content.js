import React from 'react';
import Filter from './Filter';
import MovieItem from './MovieItem';

const Content = ({ movies, records }) => {
    if(movies == null || movies.length <= 0) {
        throw new Error('No Movies Found!');
    }

    return (
        <React.Fragment>
            <Filter movieCount={records} />
            <main className="moviedata-container"> {
                movies.map(movie => <MovieItem key={movie.id} movieItem={movie} />)
            }
            </main>
        </React.Fragment>
    )
}

export default Content;