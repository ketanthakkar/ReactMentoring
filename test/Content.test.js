import React from 'react';
import { mount, shallow } from 'enzyme';
import Content from "../src/components/Content";
import Filter from "../src/components/Filter";

const movieData = [{
    "id": 337167,
    "title": "Fifty Shades Freed",
    "tagline": "Don't miss the climax",
    "vote_average": 6.1,
    "vote_count": 1195,
    "release_date": "2018-02-07",
    "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
    "budget": 55000000,
    "revenue": 136906000,
    "genres": [
    "Drama",
    "Romance"
    ],
    "runtime": 106
}];

describe('should render Content component', () => {
    const tree = shallow(<Content movies={movieData} records={10} />);
    test('Snapshot test', () => {   
        expect(tree).toMatchSnapshot();
    });

    test('should render Content component with number of records', () => {
        const count = 10;
        const tree = mount(<Content movies={movieData} records={count} />);
        expect(tree.find(Filter).prop('movieCount')).toEqual(count);
    });
});



