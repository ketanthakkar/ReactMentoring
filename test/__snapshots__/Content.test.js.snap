// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Content component Snapshot test 1`] = `
<Content
  movies={
    Array [
      Object {
        "budget": 55000000,
        "genres": Array [
          "Drama",
          "Romance",
        ],
        "id": 337167,
        "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
        "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
        "release_date": "2018-02-07",
        "revenue": 136906000,
        "runtime": 106,
        "tagline": "Don't miss the climax",
        "title": "Fifty Shades Freed",
        "vote_average": 6.1,
        "vote_count": 1195,
      },
    ]
  }
  records={10}
/>
`;
